### ISSUE TITLE
replace FPN with  RFP

### ISSUE THREAD

**Issue by yzk-lab** [joe-siyuan-qiao/DetectoRS#79]
I have a model which use  FPN,but your Recusion Feature Pyramid is better than FPN ,so i transform the RFP code into my code mmdet/neck /rfp.py,however,there are something wrong.if I just change the type in the config,when i train the model ,it occurs that  register_module() missing 1 required positional argument: 'cls',I don‘t know why.i hope you can kindly tell me the solution.

**Comment by cymdhx** [comment]
> I have a model which use FPN,but your Recusion Feature Pyramid is better than FPN ,so i transform the RFP code into my code mmdet/neck /rfp.py,however,there are something wrong.if I just change the type in the config,when i train the model ,it occurs that register_module() missing 1 required positional argument: 'cls',I don‘t know why.i hope you can kindly tell me the solution.

I want to know where the author's code about recursive FPN is. I can't find it. Thank you

**Comment by yzk-lab** [comment]
mmdet/models/necks/RFP.py




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "joe-siyuan-qiao/DetectoRS"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年4月22日(星期四) 中午1:34
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [joe-siyuan-qiao/DetectoRS] replace FPN with  RFP (#79)





  
I have a model which use FPN,but your Recusion Feature Pyramid is better than FPN ,so i transform the RFP code into my code mmdet/neck /rfp.py,however,there are something wrong.if I just change the type in the config,when i train the model ,it occurs that register_module() missing 1 required positional argument: 'cls',I don‘t know why.i hope you can kindly tell me the solution.
  
I want to know where the author's code about recursive FPN is. I can't find it. Thank you

**Comment by cymdhx** [comment]
![image](https://user-images.githubusercontent.com/73769530/115661505-1c378f80-a370-11eb-95ab-1d9038c47321.png)
Is it hrfp？ sorry I can't find rfp

**Comment by yzk-lab** [comment]
no,hrfpn means high resolution fpn,you may need to find the recursive FPN in mmdetection/models/necks




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "joe-siyuan-qiao/DetectoRS"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年4月22日(星期四) 中午1:39
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [joe-siyuan-qiao/DetectoRS] replace FPN with  RFP (#79)





 

 Is it hrfp？

**Comment by cymdhx** [comment]
> no,hrfpn means high resolution fpn,you may need to find the recursive FPN in mmdetection/models/necks
> […](#)
> ------------------ 原始邮件 ------------------ 发件人: "joe-siyuan-qiao/DetectoRS" ***@***.***>; 发送时间: 2021年4月22日(星期四) 中午1:39 ***@***.***>; ***@***.******@***.***>; 主题: Re: [joe-siyuan-qiao/DetectoRS] replace FPN with RFP (#79) Is it hrfp？ — You are receiving this because you authored the thread. Reply to this email directly, view it on GitHub, or unsubscribe.

But there seems to be no RFP file in this path. Do you have any code? Can you send it to me? Thank you

**Comment by zugofn** [comment]
> > no,hrfpn means high resolution fpn,you may need to find the recursive FPN in mmdetection/models/necks
> > […](#)
> > ------------------ 原始邮件 ------------------ 发件人: "joe-siyuan-qiao/DetectoRS" _**@**_._**>; 发送时间: 2021年4月22日(星期四) 中午1:39 _**@**_.**_>; _**@**_._**_**@**_.**_>; 主题: Re: [joe-siyuan-qiao/DetectoRS] replace FPN with RFP (#79) Is it hrfp？ — You are receiving this because you authored the thread. Reply to this email directly, view it on GitHub, or unsubscribe.
> 
> But there seems to be no RFP file in this path. Do you have any code? Can you send it to me? Thank you

Maybe you can find RFP in this path ("DetectoRS/mmdet/models/detectors/rfp.py" )

**Comment by cymdhx** [comment]
> > > no,hrfpn means high resolution fpn,you may need to find the recursive FPN in mmdetection/models/necks
> > > […](#)
> > > ------------------ 原始邮件 ------------------ 发件人: "joe-siyuan-qiao/DetectoRS" _**@**_._**>; 发送时间: 2021年4月22日(星期四) 中午1:39 _**@**_.**_>; _**@**_._**_**@**_.**_>; 主题: Re: [joe-siyuan-qiao/DetectoRS] replace FPN with RFP (#79) Is it hrfp？ — You are receiving this because you authored the thread. Reply to this email directly, view it on GitHub, or unsubscribe.
> > 
> > 
> > But there seems to be no RFP file in this path. Do you have any code? Can you send it to me? Thank you
> 
> Maybe you can find RFP in this path ("DetectoRS/mmdet/models/detectors/rfp.py" )

thank you so much

**Comment by zugofn** [comment]
> > > > no,hrfpn means high resolution fpn,you may need to find the recursive FPN in mmdetection/models/necks
> > > > […](#)
> > > > ------------------ 原始邮件 ------------------ 发件人: "joe-siyuan-qiao/DetectoRS" _**@**_._**>; 发送时间: 2021年4月22日(星期四) 中午1:39 _**@**_.**_>; _**@**_._**_**@**_.**_>; 主题: Re: [joe-siyuan-qiao/DetectoRS] replace FPN with RFP (#79) Is it hrfp？ — You are receiving this because you authored the thread. Reply to this email directly, view it on GitHub, or unsubscribe.
> > > 
> > > 
> > > But there seems to be no RFP file in this path. Do you have any code? Can you send it to me? Thank you
> > 
> > 
> > Maybe you can find RFP in this path ("DetectoRS/mmdet/models/detectors/rfp.py" )
> 
> thank you so much

油阿味儿康姆

**Comment by cymdhx** [comment]
> > > > > no,hrfpn means high resolution fpn,you may need to find the recursive FPN in mmdetection/models/necks
> > > > > […](#)
> > > > > ------------------ 原始邮件 ------------------ 发件人: "joe-siyuan-qiao/DetectoRS" _**@**_._**>; 发送时间: 2021年4月22日(星期四) 中午1:39 _**@**_.**_>; _**@**_._**_**@**_.**_>; 主题: Re: [joe-siyuan-qiao/DetectoRS] replace FPN with RFP (#79) Is it hrfp？ — You are receiving this because you authored the thread. Reply to this email directly, view it on GitHub, or unsubscribe.
> > > > 
> > > > 
> > > > But there seems to be no RFP file in this path. Do you have any code? Can you send it to me? Thank you
> > > 
> > > 
> > > Maybe you can find RFP in this path ("DetectoRS/mmdet/models/detectors/rfp.py" )
> > 
> > 
> > thank you so much
> 
> 油阿味儿康姆

哈哈哈哈 还以为你是外国人
